### PR TITLE
Implemented iota_client_get_unspent_address API.

### PR DESCRIPTION
**Pull request by oopsmonk**
In some cases, we do not need a new address but an unspent address only.

### CODE REVIEW

File: cclient/api/extended/get_new_address.c [hunk 1]
```diff
@@ -156,3 +158,49 @@ retcode_t iota_client_get_new_address(iota_client_service_t const* const serv, f
...
+  if (addr_opt.security == 0 || addr_opt.security > 3) {
+    ret = RC_CCLIENT_INVALID_SECURITY;
+    log_error(client_extended_logger_id, "%s %s\n", __func__, error_2_string(ret));
+    return ret;
+  }
+
+  for (addr_index = addr_opt.start; addr_index < addr_opt.total; addr_index++) {
+    tmp_addr = iota_sign_address_gen_flex_trits(seed, addr_index, addr_opt.security);
+    if (tmp_addr) {
+      if (was_address_spent_from(serv, tmp_addr, &is_unspent) != RC_OK) {
```
Comment (thibault-martinez) [inline cclient/api/extended/get_new_address.c]: Variable should be names `is_spent`

File: cclient/api/extended/get_new_address.c [hunk 2]
```diff
@@ -156,3 +158,49 @@ retcode_t iota_client_get_new_address(iota_client_service_t const* const serv, f
...
+        log_error(client_extended_logger_id, "were_address_spent_from failed\n");
+        free(tmp_addr);
+        goto done;
+      }
+
+      if (is_unspent) {
```
Comment (thibault-martinez) [inline cclient/api/extended/get_new_address.c]: Isn't this doing the opposite of what you want ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/api/extended/get_new_address.c (hunk 1 above)
Comment (oopsmonk) [reply cclient/api/extended/get_new_address.c]: `is_unused` and `is_unspent` is a bad naming and confusing, let's change it to `is_used` and `is_spent`.

File: cclient/api/extended/get_new_address.c (hunk 2 above)
Comment (oopsmonk) [reply cclient/api/extended/get_new_address.c]: it's exactly the functionality.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**